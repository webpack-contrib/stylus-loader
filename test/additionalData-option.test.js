const {
  compile,
  getCodeFromBundle,
  getCompiler,
  getErrors,
  getWarnings,
} = require("./helpers");

describe('"additionalData" option', () => {
  it("should work as a string", async () => {
    const testId = "./additional-data.styl";
    const additionalData = "color = coral";
    const compiler = getCompiler(testId, { additionalData });
    const stats = await compile(compiler);
    const codeFromBundle = getCodeFromBundle(stats, compiler);

    expect(codeFromBundle.css).toMatchSnapshot("css");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });

  it("should as a function", async () => {
    const testId = "./additional-data.styl";
    const additionalData = (content, loaderContext) => {
      const { resourcePath, rootContext } = loaderContext;
      // eslint-disable-next-line global-require
      const relativePath = require("path").relative(rootContext, resourcePath);

      return `
/* RelativePath: ${relativePath}; */

color = coral;
bg = gray;

${content}

.custom-class
  background: bg
        `;
    };
    const compiler = getCompiler(testId, { additionalData });
    const stats = await compile(compiler);
    const codeFromBundle = getCodeFromBundle(stats, compiler);

    expect(codeFromBundle.css).toMatchSnapshot("css");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });

  it("should as an async function", async () => {
    const testId = "./additional-data.styl";
    const additionalData = async (content, loaderContext) => {
      const { resourcePath, rootContext } = loaderContext;
      // eslint-disable-next-line global-require
      const relativePath = require("path").relative(rootContext, resourcePath);

      return `
/* RelativePath: ${relativePath}; */

color = coral;
bg = gray;

${content}

.custom-class
  background: bg
        `;
    };
    const compiler = getCompiler(testId, { additionalData });
    const stats = await compile(compiler);
    const codeFromBundle = getCodeFromBundle(stats, compiler);

    expect(codeFromBundle.css).toMatchSnapshot("css");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });
});
