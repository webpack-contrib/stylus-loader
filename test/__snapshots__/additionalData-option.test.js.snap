// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"additionalData" option should as a function: css 1`] = `
"/* RelativePath: additional-data.styl; */
body {
  color: #ff7f50;
}
.custom-class {
  background: #808080;
}
"
`;

exports[`"additionalData" option should as a function: errors 1`] = `[]`;

exports[`"additionalData" option should as a function: warnings 1`] = `[]`;

exports[`"additionalData" option should as an async function: css 1`] = `
"/* RelativePath: additional-data.styl; */
body {
  color: #ff7f50;
}
.custom-class {
  background: #808080;
}
"
`;

exports[`"additionalData" option should as an async function: errors 1`] = `[]`;

exports[`"additionalData" option should as an async function: warnings 1`] = `[]`;

exports[`"additionalData" option should work as a string: css 1`] = `
"body {
  color: #ff7f50;
}
"
`;

exports[`"additionalData" option should work as a string: errors 1`] = `[]`;

exports[`"additionalData" option should work as a string: warnings 1`] = `[]`;
